// discord.js-commando/src/util.js
function disambiguation(items, label, property = 'name') {
	const itemList = items.map(item => `"${(property ? item[property] : item).replace(/ /g, '\xa0')}"`).join(',   ');
	return `Multiple ${label} found, please be more specific: ${itemList}`;
}

function paginate(items, page = 1, pageLength = 10) {
	const maxPage = Math.ceil(items.length / pageLength);
	if(page < 1) page = 1;
	if(page > maxPage) page = maxPage;
	const startIndex = (page - 1) * pageLength;
	return {
		items: items.length > pageLength ? items.slice(startIndex, startIndex + pageLength) : items,
		page,
		maxPage,
		pageLength
	};
}

const permissions = {
	ADMINISTRATOR: 'Administrator',
	VIEW_AUDIT_LOG: 'View audit log',
	MANAGE_GUILD: 'Sunucuyu Yönet',
	MANAGE_ROLES: 'Rolleri Yönet',
	MANAGE_CHANNELS: 'Kanalları Yönet',
	KICK_MEMBERS: 'Kick members',
	BAN_MEMBERS: 'Ban members',
	CREATE_INSTANT_INVITE: 'Create instant invite',
	CHANGE_NICKNAME: 'Change nickname',
	MANAGE_NICKNAMES: 'Manage nicknames',
	MANAGE_EMOJIS: 'Emojileri Yönet',
	MANAGE_WEBHOOKS: 'Weebhookları Yönet',
	VIEW_CHANNEL: 'Read text channels and see voice channels',
	SEND_MESSAGES: 'Send messages',
	SEND_TTS_MESSAGES: 'Send TTS messages',
	MANAGE_MESSAGES: 'Mesajları Yönet',
	EMBED_LINKS: 'Embed links',
	ATTACH_FILES: 'Attach files',
	READ_MESSAGE_HISTORY: 'Read message history',
	MENTION_EVERYONE: 'Mention everyone',
	USE_EXTERNAL_EMOJIS: 'Use external emojis',
	ADD_REACTIONS: 'Tepki Ekle',
	CONNECT: 'Bağlan',
	SPEAK: 'Konuş',
	MUTE_MEMBERS: 'Kullanıcıları Sustur',
	DEAFEN_MEMBERS: 'Üyeleri Sağırlaştır',
	MOVE_MEMBERS: 'Üyeleri Taşı',
	USE_VAD: 'Use voice activity'
};

module.exports = {
	disambiguation,
	paginate,
	permissions
};
